Missing resources in Jakarta Enterprise Bean Platform tests

Three are missing resources in some of the Jakarta Enterprise Bean Platform tests.

The following seem to be missing a file, or files, named `foo.txt`. These files seem to be generated in Jakarta EE 10 and the content of the file seems to simply be `foo.txt` from looking at https://github.com/jakartaee/platform-tck/blob/1899a25b8a4c71f84cd9dd3160ff0c88a5ad108c/src/com/sun/ts/tests/ejb30/assembly/librarydirectory/custom/build.xml#L40 and https://github.com/jakartaee/platform-tck/blob/1899a25b8a4c71f84cd9dd3160ff0c88a5ad108c/src/com/sun/ts/tests/ejb30/assembly/librarydirectory/custom/build.xml#L85-L88.

- `com.sun.ts.tests.ejb30.assembly.librarydirectory.custom.ClientTest`
- `com.sun.ts.tests.ejb30.assembly.librarydirectory.defaultname.ClientTest`
- `com.sun.ts.tests.ejb30.assembly.librarydirectory.disable.ClientTest`

Some additional files that need to be created for EJB 3.0:
- `com/sun/ts/tests/ejb30/assembly/librarydirectory/custom/second-level-jar.txt`

Correct. This is the error I see:
```
[ERROR] com.sun.ts.tests.ejb30.assembly.librarydirectory.custom.ClientTest -- Time elapsed: 2.842 s <<< ERROR!
java.lang.RuntimeException: Could not invoke deployment method: public static org.jboss.shrinkwrap.api.spec.EnterpriseArchive com.sun.ts.tests.ejb30.assembly.librarydirectory.custom.ClientTest.createDeployment(tck.arquillian.porting.lib.spi.TestArchiveProcessor)
	at org.jboss.arquillian.container.test.impl.client.deployment.AnnotationDeploymentScenarioGenerator.invoke(AnnotationDeploymentScenarioGenerator.java:165)
	at org.jboss.arquillian.container.test.impl.client.deployment.AnnotationDeploymentScenarioGenerator.generateDeploymentContent(AnnotationDeploymentScenarioGenerator.java:119)
	at org.jboss.arquillian.container.test.impl.client.deployment.AnnotationDeploymentScenarioGenerator.generateDeploymentContent(AnnotationDeploymentScenarioGenerator.java:58)
	at org.jboss.arquillian.container.test.impl.client.deployment.AbstractDeploymentScenarioGenerator.generate(AbstractDeploymentScenarioGenerator.java:30)
	at org.jboss.arquillian.container.test.impl.client.deployment.DeploymentGenerator.generateDeployment(DeploymentGenerator.java:84)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:102)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:89)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:61)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.execute(ContainerEventController.java:94)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:102)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:89)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:82)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:94)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:68)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:85)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:94)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.beforeClass(EventTestRunnerAdaptor.java:88)
	at org.jboss.arquillian.junit5.ArquillianExtension.beforeAll(ArquillianExtension.java:41)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.jboss.arquillian.container.test.impl.client.deployment.AnnotationDeploymentScenarioGenerator.invoke(AnnotationDeploymentScenarioGenerator.java:163)
	... 29 more
Caused by: java.lang.IllegalArgumentException: Resource should be specified
	at org.jboss.shrinkwrap.impl.base.Validate.notNull(Validate.java:43)
	at org.jboss.shrinkwrap.impl.base.container.ContainerBase.addAsResource(ContainerBase.java:1080)
	at com.sun.ts.tests.ejb30.assembly.librarydirectory.custom.ClientTest.createDeployment(ClientTest.java:168)
	... 31 more
```

--- a/tcks/apis/enterprise-beans/ejb30/src/main/java/com/sun/ts/tests/ejb30/assembly/librarydirectory/custom/ClientTest.java
+++ b/tcks/apis/enterprise-beans/ejb30/src/main/java/com/sun/ts/tests/ejb30/assembly/librarydirectory/custom/ClientTest.java
@@ -127,12 +127,10 @@ public class ClientTest extends com.sun.ts.tests.ejb30.assembly.librarydirectory
                 JavaArchive lib_shared_lib = ShrinkWrap.create(JavaArchive.class, "lib-shared.jar");
 
                 // The resources
-                        libURL = Client.class.getResource("/com/sun/ts/tests/ejb30/assembly/common/foo.txt");
-                        lib_shared_lib.addAsResource(libURL, "/com/sun/ts/tests/ejb30/assembly/common/foo.txt");
-                        libURL = Client.class.getResource("/com/sun/ts/tests/ejb30/assembly/librarydirectory/custom/foo.txt");
-                        lib_shared_lib.addAsResource(libURL, "/com/sun/ts/tests/ejb30/assembly/librarydirectory/custom/foo.txt");
-                        libURL = Client.class.getResource("/foo.txt");
-                        lib_shared_lib.addAsResource(libURL, "/foo.txt");
+                StringAsset fooTxt = new StringAsset("foo.txt");
+                lib_shared_lib.addAsResource(fooTxt, "/com/sun/ts/tests/ejb30/assembly/common/foo.txt");
+                lib_shared_lib.addAsResource(fooTxt, "/com/sun/ts/tests/ejb30/assembly/librarydirectory/custom/foo.txt");
+                lib_shared_lib.addAsResource(fooTxt, "/foo.txt");
 
                 ejb3_assembly_librarydirectory_custom_ear.addAsLibrary(lib_shared_lib);
                 JavaArchive shared_lib = ShrinkWrap.create(JavaArchive.class, "shared.jar");

--- a/tcks/apis/enterprise-beans/ejb30/src/main/java/com/sun/ts/tests/ejb30/assembly/librarydirectory/defaultname/ClientTest.java
+++ b/tcks/apis/enterprise-beans/ejb30/src/main/java/com/sun/ts/tests/ejb30/assembly/librarydirectory/defaultname/ClientTest.java
@@ -142,8 +142,8 @@ public class ClientTest extends com.sun.ts.tests.ejb30.assembly.librarydirectory
                 JavaArchive sub_shared_lib = ShrinkWrap.create(JavaArchive.class, "sub-shared.jar");
 
                 // The resources
-                        libURL = Client.class.getResource("/com/sun/ts/tests/ejb30/assembly/librarydirectory/defaultname/foo.txt");
-                        sub_shared_lib.addAsResource(libURL, "/com/sun/ts/tests/ejb30/assembly/librarydirectory/defaultname/foo.txt");
+                StringAsset fooTxt = new StringAsset("foo.txt");
+                sub_shared_lib.addAsResource(fooTxt, "/com/sun/ts/tests/ejb30/assembly/librarydirectory/defaultname/foo.txt");
 
                 ejb3_assembly_librarydirectory_defaultname_ear.addAsLibrary(sub_shared_lib);
                 JavaArchive hello_client_view_lib = ShrinkWrap.create(JavaArchive.class, "hello-client-view.jar");

--- a/tcks/apis/enterprise-beans/ejb30/src/main/java/com/sun/ts/tests/ejb30/assembly/librarydirectory/disable/ClientTest.java
+++ b/tcks/apis/enterprise-beans/ejb30/src/main/java/com/sun/ts/tests/ejb30/assembly/librarydirectory/disable/ClientTest.java
@@ -165,12 +165,10 @@ public class ClientTest extends com.sun.ts.tests.ejb30.assembly.librarydirectory
 
 
                 // The resources
-                        libURL = Client.class.getResource("/com/sun/ts/tests/ejb30/assembly/common/foo.txt");
-                        shared_lib.addAsResource(libURL, "/com/sun/ts/tests/ejb30/assembly/common/foo.txt");
-                        libURL = Client.class.getResource("/com/sun/ts/tests/ejb30/assembly/librarydirectory/disable/foo.txt");
-                        shared_lib.addAsResource(libURL, "/com/sun/ts/tests/ejb30/assembly/librarydirectory/disable/foo.txt");
-                        libURL = Client.class.getResource("/foo.txt");
-                        shared_lib.addAsResource(libURL, "/foo.txt");
+                StringAsset fooTxt = new StringAsset("foo.txt");
+                shared_lib.addAsResource(fooTxt, "/com/sun/ts/tests/ejb30/assembly/common/foo.txt");
+                shared_lib.addAsResource(fooTxt, "/com/sun/ts/tests/ejb30/assembly/librarydirectory/disable/foo.txt");
+                shared_lib.addAsResource(fooTxt, "/foo.txt");
 
                 ejb3_assembly_librarydirectory_disable_ear.addAsLibrary(shared_lib);
 

--- a/tcks/apis/enterprise-beans/ejb30/src/main/java/com/sun/ts/tests/ejb30/timer/basic/mdb/ClientTest.java
+++ b/tcks/apis/enterprise-beans/ejb30/src/main/java/com/sun/ts/tests/ejb30/timer/basic/mdb/ClientTest.java
@@ -104,6 +104,9 @@ public class ClientTest extends com.sun.ts.tests.ejb30.timer.basic.mdb.Client {
             if(ejbResURL != null) {
               ejb3_timer_basic_mdb_ejb.addAsManifestResource(ejbResURL, "sun-ejb-jar.xml");
             }
+            StringAsset fooTxt = new StringAsset("foo.txt");
+            ejb3_timer_basic_mdb_ejb.addAsResource(fooTxt, "com/sun/ts/tests/ejb30/timer/basic/mdb/foo.txt");
+
             // Call the archive processor
             archiveProcessor.processEjbArchive(ejb3_timer_basic_mdb_ejb, Client.class, ejbResURL);
 

